### DISCUSSION TITLE
Questions after completed Foundry Full Course

### DISCUSSION THREAD

**Discussion by terranweb3** [Cyfrin/foundry-full-course-cu#909] [General]
Hello everyone, after completed the Foundry course, should I continue studying Hardhat? Previously, I worked as a Full Stack Developer, so I know about TypeScript, and I can make a complete Web3 application. I have seen that Hardhat uses TypeScript for testing and scripting. Should I learn or should I just practice now by developing Web3 projects with the Foundry library?

- Thanks for your answer!

**Comment by borhanularif** [comment]
Have you completed the Foundry course?

**Reply by terranweb3** [reply]
I'm completed that course

**Reply by terranweb3** [reply]
Sorry for my grammatically incorrect question

**Reply by theirrationalone** [reply]
Typescript is really evolving, we should learn it. As you said you have completed both courses. 

IMO, You should continue to a solidity oriented learning approach. 

You should continue Building contracts, make heavy use of foundry as a tool, create fullstack applications. 

You should now learn what defi is?? To the darkest deep. 

You should scratch EIPs i.e., ERC20, ERC721, Proxies etc. 

🚀🚀 🚀🚀🚀🚀🚀🚀🚀

**Reply by terranweb3** [reply]
@theirrationalone thanks for your advice <3

**Comment by cromewar** [comment]
@terrancrypt our upcoming video is exactly about this, we are going to talk about, what's next?  we are going to give you the "I finished Patrick's course now what?" roadmap. Will be out soon!

**Reply by theirrationalone** [reply]
🤤🤤🤩🤩🤩, Pinned, Tuned ;`)

**Reply by terranweb3** [reply]
Thank you a lot ! I love it <3
